docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+
+# Use the context to resolve as many of the template fields as possible.
+fields = context.as_template_fields(template)
+
+# Manually resolve the remaining fields that can't be figured out automatically from context.
+fields["name"] = "myscene"
+
+# Get an authenticated Shotgun API instance from the engine.
+sg = current_engine.shotgun
+# Run a Shotgun API query to summarize the maximum version number on PublishedFiles that
+# are linked to the task and match the provided name.
+# Since PublishedFiles generated by the Publish app have the extension on the end of the name we need to add the
+# extension in our filter.
+r = sg.summarize(entity_type="PublishedFile",
+                 filters = [["task", "is", {"type":"Task", "id": context.task["id"]}],
+                            ["name","is", fields["name"] + ".ma"]],
+                 summary_fields=[{"field":"version_number", "type":"maximum"}])
+
+# Extract the version number and add 1 to it.
+# In scenarios where there are no files already this summary will return 0.
+fields["version"] = r["summaries"]["version_number"] + 1
+
+# Use the fields to resolve the template path into an absolute path.
+publish_path = template.apply_fields(fields)
+
+# Make sure we create any missing folders.
+current_engine.ensure_folder_exists(os.path.dirname(publish_path))
+
+# Create a empty file on disk. (optional - should be replaced by actual file save or copy logic)
+sgtk.util.filesystem.touch_file(publish_path)
+```
+
+## Part 7 - Registering a published file
+
+Now that you have a path you're ready to publish it. To do this we can use the utility method
+[`sgtk.util.register_publish()` method](https://developer.shotgunsoftware.com/tk-core/utils.html?#sgtk.util.register_publish).
+
+It is possible to use the Shotgun API's `create` method to create a `PublishedFile` entity as well, but we strongly
+recommend using the Toolkit API for this as it will ensure all the required fields are provided and filled in correctly.
+
+```python
+# So as to match the publish app's default behavior, we are adding the extension to the end of the publish name.
+# This is optional, however.
+publish_name = fields["name"] + ".ma"
+version_number = fields["version"]
+
+# Now register the publish
+sgtk.util.register_publish(tk,
+                           context,
+                           publish_path,
+                           publish_name,
+                           version_number,
+                           published_file_type = "Maya Scene")
+```
+
+At this point, it's also worth noting that our [Publish app](https://support.shotgunsoftware.com/hc/en-us/articles/115000097513-Publishing-your-work)
+also comes with [its own API](https://developer.shotgunsoftware.com/tk-multi-publish2/) as well. 
+Although that is still essentially using this same `register_publish` method. 
+It builds upon the publishing process by providing a framework to handle collection, validation, and publishing.
```

Comment:
- This line and the one above should be one sentence, separated by a comma. 
 - I'd add `()` to `register_publish`